File: backend/test_login.js
const axios = require('axios');

async function testLogin(username) {
  try {
    console.log(`\n=== 测试前端模拟登录: ${username} ===`);
    const response = await axios.post('http://127.0.0.1:56793/api/auth/login', 
      { username, password: '123456' },
      { timeout: 5000 }
    );
    console.log('Response data keys:', Object.keys(response.data));
    console.log('Has token:', !!response.data.token);
    console.log('Has user:', !!response.data.user);
    if (response.data.user) {
      console.log('User role:', response.data.user.role);
      console.log('User username:', response.data.user.username);
    }
    return response.data;
  } catch (error) {
    console.log('Error:', error.message);
    if (error.response) {
      console.log('Status:', error.response.status);
      console.log('Data:', error.response.data);
    }
    return null;
  }
}

(async () => {
  for (const u of ['admin', 'platform', 'ops', 'stationmaster', 'zhangsan']) {
    await testLogin(u);
  }
})();

File: api/.test_login.py
import requests, json
BASE = "http://localhost:3000/api"

r = requests.post(BASE + "/auth/login", json={"username":"courier1","password":"courier123"})
d = r.json()
print("Courier1 login:", d["code"], d["data"]["user"]["name"], d["data"]["user"]["role"])
ct = d["data"]["token"]

r = requests.get(BASE + "/auth/me", headers={"Authorization": "Bearer " + ct})
d = r.json()
print("auth/me:", d["data"]["name"], d["data"]["role"])

r = requests.post(BASE + "/auth/login", json={"username":"admin1","password":"admin123"})
d = r.json()
print("Admin1 login:", d["code"], d["data"]["user"]["name"], d["data"]["user"]["role"])

r = requests.post(BASE + "/auth/login", json={"username":"operator1","password":"operator123"})
d = r.json()
print("Operator1 login:", d["code"], d["data"]["user"]["name"], d["data"]["user"]["role"])

r = requests.post(BASE + "/auth/login", json={"username":"courier1","password":"wrong"})
d = r.json()
print("Wrong password:", d["code"], d["message"])
